packages/editor/src/hooks/block-bindings.js

```diff
@@ -0,0 +1,303 @@
+/**
+ * External dependencies
+ */
+import clsx from 'clsx';
+
+/**
+ * Internal dependencies
+ */
+import {
+	canBindAttribute,
+	getBindableAttributes,
+} from './use-bindings-attributes';
+import { store as editorStore } from '../store';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	MenuGroup,
+	MenuItem,
+	__experimentalToolsPanel as ToolsPanel,
+	__experimentalToolsPanelItem as ToolsPanelItem,
+	__experimentalHStack as Hstack,
+	__experimentalTruncate as Truncate,
+	__experimentalItemGroup as ItemGroup,
+	__experimentalDropdownContentWrapper as DropdownContentWrapper,
+	Dropdown,
+	Button,
+	Icon,
+} from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { chevronRightSmall, customPostType } from '@wordpress/icons';
+import {
+	InspectorControls,
+	store as blockEditorStore,
+} from '@wordpress/block-editor';
+import { addFilter } from '@wordpress/hooks';
+import {
+	createHigherOrderComponent,
+	useViewportMatch,
+} from '@wordpress/compose';
+
+const popoverProps = {
+	placement: 'left-start',
+	offset: 36,
+	shift: true,
+	className: 'block-editor-block-bindings__popover',
+	headerTitle: __( 'Custom Fields' ),
+};
+
+const useToolsPanelDropdownMenuProps = () => {
+	const isMobile = useViewportMatch( 'medium', '<' );
+	return ! isMobile
+		? {
+				popoverProps: {
+					placement: 'left-start',
+					// For non-mobile, inner sidebar width (248px) - button width (24px) - border (1px) + padding (16px) + spacing (20px)
+					offset: 259,
+				},
+		  }
+		: {};
+};
+
+function BlockBindingsPanelDropdown( { postMeta, addConnection, attribute } ) {
+	return (
+		<DropdownContentWrapper paddingSize="small">
+			<MenuGroup label={ __( 'Custom Fields' ) }>
```

Comment:
Have you considered using [`DropdownMenu`](https://wordpress.github.io/gutenberg/?path=/docs/components-dropdownmenu--docs) instead? It should also render with the correct borders and shadow variants automatically when used inside a `Toolbar`

Reply:
I think I had issues with it, as it was in https://github.com/WordPress/gutenberg/pull/62880/commits/9445effed690b4ced4b162a70f6d2e2cd16510f4#diff-f9c2cc67b161615938447e7ad71c47ce49a06d58639477d3b9b7551a9fb1c497L96

I'll re-check

Reply:
Can a DropdownMenu v1 have a ToolPanel Item as a trigger or only an icon?

Reply:
It should be — see the [default Storybook example](https://wordpress.github.io/gutenberg/?path=/docs/components-dropdownmenu--docs). Storybook examples are the best way to familiarize with a component and its APIs.

If you're rendering the dropdown menu in a `Toolbar`, you may want to use `ToolbarDropdownMenu ` instead (you can see it in action in the [`Toolbar` default Storybook example](https://wordpress.github.io/gutenberg/?path=/docs/components-toolbar--docs))

Reply:
(I initially misinterpreted the question, but I now updated the initial reply, hopefully it addresses your questions better!)